src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.hpp

```diff
@@ -170,8 +174,8 @@ class FlightTaskAutoFollowTarget : public FlightTask
 	follow_target_estimator_s _follow_target_estimator;
 	matrix::Vector2f _target_velocity_unit_vector;
 
-	// Lowpass filters for smoothingtarget position because it's used for setpoint generation
-	AlphaFilter<matrix::Vector3f> _target_position_filter;
+	// Second Order Filter to calculate kinematically feasible target position
+	SecondOrderReferenceModel<matrix::Vector3f> _target_position_filter;
```

Comment:
I would recommend naming it `_target_pose_filter`, as it is not just filtering position, but also velocity.